eks-worker-al2.json

```diff
@@ -13,10 +14,10 @@
     "aws_session_token": "{{env `AWS_SESSION_TOKEN`}}",
     "binary_bucket_name": "amazon-eks",
     "binary_bucket_region": "us-west-2",
-    "cni_plugin_version": "v0.8.6",
-    "containerd_version": "1.6.6-1.amzn2.0.2",
+    "cni_plugin_version": "",
+    "containerd_version": "",
     "creator": "{{env `USER`}}",
-    "docker_version": "20.10.17-1.amzn2.0.1",
+    "docker_version": "",
```

Comment:
Presumably, if you set these, it would override the variables set below?

Reply:
I have to iron out the order of overrides; I think the var file would actually override this value. My guess is it goes:
1. `variables` section in the template.
2. Anything in `-var-file`
3. Anything defined with `-var`.

Reply:
Looks like it's probably better to define these version variables in the template with a value of `null`, so they have to be defined either in the `version-variables.json` file or with a `make` argument.
> If the default value is null, then the user variable will be required. This means that the user must specify a value for this variable or template validation will fail.
https://developer.hashicorp.com/packer/docs/templates/legacy_json_templates/user-variables

Reply:
Can we remove the properties/variables altogether from eks-worker-al2.json to avoid this confusion?

Reply:
I think it's better to have all the template variables called out in the template itself (and this is a requirement of HCL-format templates, which we should eventually move to).

This change also relies on them being defined in the template, and is preferable to manually maintaining this list (which has been unreliable in the past): https://github.com/awslabs/amazon-eks-ami/pull/1079/files#diff-76ed074a9305c04054cdebb9e9aad2d818052b07091de1f20cad0bbac34ffb52R2

Reply:
I'm fine with this approach. My only concern is the priority order. IMO, command line args should override everything, `eks-worker-al2.json` should override the defaults, and this new file should just include default values that get set if nothing overrides it.

Reply:
> eks-worker-al2.json should override the defaults

The template file *is* the default. Changing the value there to `null` means the variable has to be defined elsewhere.

The variable file overrides the template. In practice, these will be the "defaults" for our users. Users can provide their own variable file with the `PACKER_VARIABLE_FILE` make argument.

The `-var` flag overrides everything. Users can override *our* variable file with these, if they don't want to provide their own variable file.